cpp/test/IceGrid/deployer/AllTests.cpp

```diff
@@ -50,29 +49,28 @@ bool isLongLineEnd(const string& line)
 
 }
 
-struct ProxyIdentityEqual : public std::binary_function<Ice::ObjectPrx,string,bool>
+struct ProxyIdentityEqual
```

Comment:
Should use Ice::ProxyIdentityEqual, defined in Ice/Proxy.h

Reply:
This `ProxyIdentityEqual` is comparing a proxy object with a string.

Reply:
I think with C++11 you can just use a lambda with find_if don't need for a functor

Reply:
We use this in many places, copying the lambda wouldn't be as good.

Reply:
I was thinking something like 
```
identityEqual(s)
{
    auto id = Ice::stringToIdentity(s);
    return [id](p)
    {
        return p->ice_getIdentity() == id;
    }
}
test(find_if(objs.begin(), objs.end(), identityEqual("Server1")) != objs.end());
```
But I didn't test it

Reply:
Oh, nice, I'll try that!

Reply:
That worked great!